lib/vision/index.js

```diff
@@ -0,0 +1,1379 @@
+ *
+ * To learn more about the Vision API, see
+ * [Getting Started](https://cloud.google.com/vision/docs/getting-started).
+ *
+ * @resource [Getting Started]{@link https://cloud.google.com/vision/docs/getting-started}
+ * @resource [Image Best Practices]{@link https://cloud.google.com/vision/docs/image-best-practices}
+ *
+ * @param {object} options - [Configuration object](#/docs).
+ *
+ * @example
+ * var gcloud = require('gcloud')({
+ *   keyFilename: '/path/to/keyfile.json',
+ *   projectId: 'grape-spaceship-123'
+ * });
+ *
+ * var vision = gcloud.vision();
+ */
+function Vision(options) {
+  if (!(this instanceof Vision)) {
+    options = util.normalizeArguments(this, options);
+    return new Vision(options);
+  }
+
+  var config = {
+    baseUrl: 'https://vision.googleapis.com/v1',
+    projectIdRequired: false,
+    scopes: [
+      'https://www.googleapis.com/auth/cloud-platform'
+    ]
+  };
+
+  Service.call(this, config, options);
+}
+
+nodeutil.inherits(Vision, Service);
+
+Vision.likelihood = {
+  VERY_UNLIKELY: VERY_UNLIKELY,
+  UNLIKELY: UNLIKELY,
+  POSSIBLE: POSSIBLE,
+  LIKELY: LIKELY,
+  VERY_LIKELY: VERY_LIKELY
+};
+
+/**
+ * Run image detection and annotation for an image or batch of images.
+ *
+ * This is an advanced API method that requires raw
+ * [`AnnotateImageRequest`](https://cloud.google.com/vision/reference/rest/v1/images/annotate#AnnotateImageRequest)
+ * objects to be provided. If that doesn't sound like what you're looking for,
+ * you'll probably appreciate {module:vision#detect}.
+ *
+ * @resource [images.annotate API Reference]{@link https://cloud.google.com/vision/reference/rest/v1/images/annotate}
+ *
+ * @param {object|object[]} requests - An `AnnotateImageRequest` or array of
+ *     `AnnotateImageRequest`s. See an
+ *     [`AnnotateImageRequest`](https://cloud.google.com/vision/reference/rest/v1/images/annotate#AnnotateImageRequest).
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {object} callback.apiResponse - Raw API response.
+ *
+ * @example
+ * var annotateImageReq = {
+ *   // See the link in the parameters for `AnnotateImageRequest`.
+ * };
+ *
+ * vision.annotate(annotateImageReq, function(err, annotations, apiResponse) {
+ *   // annotations = apiResponse.responses
+ * });
+ */
+Vision.prototype.annotate = function(requests, callback) {
+  this.request({
+    method: 'POST',
+    uri: 'images:annotate',
+    json: {
+      requests: arrify(requests)
+    }
+  }, function(err, resp) {
+    if (err) {
+      callback(err, null, resp);
+      return;
+    }
+
+    callback(null, resp.responses, resp);
+  });
+};
+
+/**
+ * Detect properties from an image (or images) of one or more types.
+ *
+ * <h4>API simplifications</h4>
+ *
+ * The API will commonly return an `MID` field, which maps to a [freebase entity
+ * ID](http://wiki.freebase.com/wiki/Machine_ID). We rename this to simply `id`.
```

Comment:
If the upstream API returns the field as `MID` my instinct would be to mirror that as `mid`. I think if a user were to cross-reference the upstream docs, this would make comparing the two a little easier.